Fix query parameters and add function

wpoely86 opened the pull request:
This builds on top of #23 

It makes that you can actually to a proper GET call with parameters. Until now, they were ignored.

I've also added the `get_running_jobs` which needs api_version 32

vaxvms commented:
Seems good to me. 

Given API version bump, rundeck 3.1.0 is required unless you specify the api version in the constructor.
Maybe it should be written somewhere ?

wpoely86 commented:
@vaxvms I've added it in the `README.md`?

vaxvms commented:
@pschmitt can you review our pull requests ?

pschmitt commented:
Since this incorporates #23 i'll close that one.

pschmitt commented on setup.py:
```diff
@@ -5,7 +5,7 @@
 setup(
     name='pyrundeck',
-    version='0.9.7',
+    version='0.9.8',
```
Please don't bump version in PRs. I'm doing that.

pschmitt commented on pyrundeck/rundeck.py:
```diff
@@ -19,7 +19,7 @@
 class Rundeck():
     def __init__(self, rundeck_url, token=None, username=None, password=None,
-                 api_version=18, verify=True):
+                 api_version=32, verify=True):
```
Why update the default? Does api_version=18 not work for Rundeck 3+?

wpoely86 replied:
Because I added `get_running_jobs` which requires api_version 32.

I leave it like on 18 and put a warning on that function (but rundeck itself will also give an error)

pschmitt commented on README.md:
```diff
@@ -2,6 +2,31 @@
 
 This is a Python REST API client for Rundeck 2.6+
 
+The current version needs Rundeck 3.1+ but many parts still work
```
I'm not a fan of requiring Rundeck 3+
Backward compatibility is a must for me.

wpoely86 replied:
Sure, whatever you want. It's only a single function that only works with 3.1

pschmitt replied:
This line is confusing, especially since it follows "REST API client for Rundeck 2.6+". Maybe something like: "some functions are only available on Rundeck 3+"?

pschmitt commented on pyrundeck/rundeck.py:
```diff
@@ -33,7 +33,7 @@ def auth(self):
         r = requests.post(
             url,
-            params=p,
+            data=p,
```
Does this work for Rundeck 2.x?

wpoely86 replied:
I haven't test it on Rundeck 2.x and the documentation is not clear on it.

pschmitt replied:
I tested this against my Rundeck 2 instance. It works. It's settled! :ok_man:

wpoely86 commented:
@pschmitt I've tried addressing all your remarks.

I've also slightly changed how the auth works. Currently a auth cookie is requested in all cases and always passed. If you give it a token, this is not needed.

wpoely86 commented:
@pschmitt Are you planning to do a version bump any time soon?

pschmitt commented:
https://pypi.org/project/pyrundeck/0.9.8/
